### DISCUSSION TITLE
Null check operator used on a null value error in Google Sign-In integration

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#1006] [Q&A] [answered]
I'm following this article for trying out Google Sign-In integration in web app on Chrome:

https://medium.com/serverpod/integrating-google-sign-in-with-serverpod-authentication-part-2-6fade3099baf

but cannot really sign-in with Google and I saw this error in Serverpod Insights:

![Screenshot 2023-06-08 at 1 42 00 PM](https://github.com/serverpod/serverpod/assets/32797520/5f00aa88-bfe4-4080-9a77-7eb87e6fa6fb)

Anybody can share some idea to solve this issue?

**Comment by Isakdl** [comment]
What version of Serverpod are you on?

To use signing with Google on the web you have to be on 1.1.1 or greater as there is a fix for the web integration in that version.

**Reply by Frank1016** [reply]
I'm on the latest 1.1.1

**Comment by flutterindiadev** [comment]
Is there any fix to this? I am facing the same issue

**Comment by Isakdl** [comment, marked as answer]
Quoting my self from related issue that should resolve this problem.

> I just took a look at the crash, we need to have a better error message here, but the reason is that the _server_ is not loading the secret credentials.
> 
> Have you put the serverside config file in your server project?
> 
> The server credentials need to be put into the `config` folder, and has to be named `google_client_secret.json`.
> 
> [Instructions from our medium article](https://medium.com/serverpod/integrating-google-sign-in-with-serverpod-authentication-part-2-6fade3099baf):
> 
> [In the Google Cloud console](https://console.cloud.google.com)
> 
> **Create Credentials**: Navigate to ‘Credentials’ under ‘APIs & Services’. Click ‘Create Credentials’ and select ‘OAuth client ID’. Configure the OAuth client as a ‘Web application’. Name your client and leave the Authorized JavaScript origins and Authorized redirect URIs fields blank for now.
> 
> **Obtain the JSON credentials**: After creating the credentials, you’ll be provided with an option to download the secrets as a JSON. Click the “DOWNLOAD JSON” button. Rename the file to `google_client_secret.json`.
> 
> **Add secret to Serverpod**: Move the secret file into your server project and put it under `config/google_client_secret.json`. Serverpod will by default look for this file to interact with Google.



https://github.com/serverpod/serverpod/issues/1532
